### ISSUE TITLE
Support for Isolation Forest

### ISSUE THREAD

**Issue by zhanjiezhu** [microsoft/hummingbird#178]
Hi there, thanks for the great package, I would like to work on the support for Isolation Forest sklearn implementation (https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.IsolationForest.html). Since this implementation has 2 main methods "predict" and "decision_function", where the "decision_function" returns the anomaly scores, and "predict" returns either -1 (anomaly) or 1 (normal) based on comparing the anomaly score with an "offset_".

I would propose following changes in order to support Isolation Forest:
1. Replace "value" in each tree with (normalized) depth, since the existing value in each tree is just useless random numbers due to the randomness when growing each tree in Isolation Forest. To do this, I will create a separate "convert_sklearn_isolation_forest" and "_get_parameters_for_sklearn_iforest" to extract and change the tree parameters. I have already done this step and it's returning expected results. 
2. create a new "PyTorchBackendModelAnomalyDetection", in which I will implement "predict" and "decision_function", and I will also need to use the "offset_" in extra_config to normalize the final output. To be able to call "PyTorchBackendModelAnomalyDetection", I will also need to add a boolean flag "anomaly_detection" just like "regression".

I've already done the first step, but would like to get your feedback before continuing.

Thanks!

**Comment by interesaaat** [comment]
Hey @zhanjiezhu this looks great! I assigned the issue to you.
